### PR TITLE
Meson

### PR DESCRIPTION
**Pull request by Nine-H**
next gen music needs a next gen build system. currently non-functional. any assistance and pointers much appreciated. chasing dependencies right now.

Core is failing with:
```
../core/PluginManager.vala:212.33-212.37: error: The name `Build' does not exist in the context of `Noise.Plugins.Manager.new'
        engine.add_search_path (Build.PLUGIN_DIR, null);
                                ^^^^^
../core/Utils/Icon.vala:65.65-65.69: error: The name `Build' does not exist in the context of `Noise.Icon.theme'
                _theme.append_search_path (Path.build_filename (Build.ICON_DIR));
```
which I believe is related to cmake but I'm not sure.

client fails because it can't access core.

### CODE REVIEW

**Comment by Nine-H** [comment]
```
../src/main.vala:40.27-40.31: error: The name `Build' does not exist in the context of `main'
    string package_name = Build.GETTEXT_PACKAGE;
                          ^^^^^
../src/main.vala:41.48-41.52: error: The name `Build' does not exist in the context of `main'
    string langpack_dir = Path.build_filename (Build.DATADIR, "locale");
                                               ^^^^^
../src/LocalBackend/LocalSmartPlaylist.vala:37.5-37.24: warning: Noise.LocalSmartPlaylist._name hides inherited field `Noise.Playlist._name'. Use the `new' keyword if hiding was intentional
    private string _name = null;
    ^^^^^^^^^^^^^^^^^^^^
../src/LocalBackend/LocalStaticPlaylist.vala:35.5-35.24: warning: Noise.LocalStaticPlaylist._name hides inherited field `Noise.Playlist._name'. Use the `new' keyword if hiding was intentional
    private string _name = null;
    ^^^^^^^^^^^^^^^^^^^^
../src/Views/Wrappers/PlaylistViewWrapper.vala:59.34-59.43: warning: Gtk.Action has been deprecated since 3.10. Use GLib.Action
../src/Noise.vala:39.26-39.30: error: The name `Build' does not exist in the context of `Noise.App'
        build_data_dir = Build.DATADIR;
                         ^^^^^
../src/Noise.vala:40.30-40.34: error: The name `Build' does not exist in the context of `Noise.App'
        build_pkg_data_dir = Build.PKG_DATADIR;
                             ^^^^^
../src/Noise.vala:41.30-41.34: error: The name `Build' does not exist in the context of `Noise.App'
        build_release_name = Build.RELEASE_NAME;
                             ^^^^^
../src/Noise.vala:42.25-42.29: error: The name `Build' does not exist in the context of `Noise.App'
        build_version = Build.VERSION;
                        ^^^^^
../src/Noise.vala:43.30-43.34: error: The name `Build' does not exist in the context of `Noise.App'
        build_version_info = Build.VERSION_INFO;
                             ^^^^^
../src/Noise.vala:45.26-45.30: error: The name `Build' does not exist in the context of `Noise.App'
        program_name = _(Build.APP_NAME);
                         ^^^^^
```

getting there

File: core/Utils/Icon.vala [hunk 1]
```diff
@@ -62,7 +62,7 @@ public class Noise.Icon : Object {
                 // This only works if Build.ICON_DIR contains a sub-directory named "hicolor"
                 // containing all the fallback icons (possibly organized into sub-folders as well),
                 // or if the icons are immediate children of this directory.
-                _theme.append_search_path (Path.build_filename (Build.ICON_DIR));
+                _theme.append_search_path (Path.build_filename ("/usr/share/noise/icons"));
```
Comment (Bat41) [inline core/Utils/Icon.vala]: `Build.ICON_DIR` should be working I think.

Comment (Nine-H) [reply core/Utils/Icon.vala]: yep, working on a python script for post install I'll get to it. not sure chucking the hardcoded paths into build.vala is going to cut it, but it works :D

**Comment by Nine-H** [comment]
does anyone know how to make the metadata validation tests pass?

File: core/build.vala [hunk 2]
```diff
@@ -2,11 +2,11 @@ namespace Build {
     public const string APP_NAME = "noise";
```
Comment (Bat41) [inline core/build.vala]: I don't really like hardcoding thinks like that. Meson offers a feature to write build info to a file: http://mesonbuild.com/Configuration.html

**Comment by Bat41** [comment]
I think you should install the `desktop-file-utils` package.

**Comment by danrabbit** [comment]
I'm going to mark this as "Blocked" because it will require packaging changes once merged. Please don't merge this branch without Cody's approval

**Comment by Bat41** [comment]
I think this PR is ready for review. cc @codygarver

**Pull request merged**